### PR TITLE
Feature/update expression

### PR DESCRIPTION
**Pull request by Iurysab**

### CODE REVIEW

File: extensions/expression.lua [hunk 1]
```diff
@@ -288,3 +288,130 @@ function Expression.add_agents_right(self, ...)
...
+    warning(tag .. ": data will lose chronology");
+
+    local inital_stage = self:inital_stage();
+    local last_stage   = self:last_stage();
+
+    local vec_selc_stag = {};
+    for stg = inital_stage,last_stage do
+        for _,stg_v in ipairs(vec_stages) do
+            if( stg == stg_v ) then
+                goto stg_found
```
Comment (raphasampaio) [inline extensions/expression.lua]: we should not use the goto command

File: extensions/expression.lua [hunk 2]
```diff
@@ -288,3 +288,130 @@ function Expression.add_agents_right(self, ...)
...
+
+    if str == nil then
+        error(tag .. ": prefix string must not be nil");
+    end
+
+    local lenth_str = string.len(str);
+
+    local found_agents = {};
+    local agents = self:agents();
+    for _,agent in ipairs(agents) do
```
Comment (raphasampaio) [inline extensions/expression.lua]: use the syntax formatter

File: extensions/expression.lua [hunk 3]
```diff
@@ -288,3 +288,130 @@ function Expression.add_agents_right(self, ...)
...
+        if (string.sub(agent, 1, lenth_str) == str) then
```
Comment (raphasampaio) [inline extensions/expression.lua]: don't need these parenthesis

File: extensions/expression.lua [hunk 4]
```diff
@@ -288,3 +288,130 @@ function Expression.add_agents_right(self, ...)
...
+            table.insert(found_agents,agent)
+        end
+    end
+    return self:select_agents(found_agents)
```
Comment (raphasampaio) [inline extensions/expression.lua]: there are some missing ;

File: extensions/expression.lua [hunk 5]
```diff
@@ -288,3 +288,130 @@ function Expression.add_agents_right(self, ...)
...
+    if vec_stages == nil then
+        error(tag .. ": stages vector must not be nil");
+    end
+
+    warning(tag .. ": data will lose chronology");
+
+    local inital_stage = self:inital_stage();
+    local last_stage   = self:last_stage();
+
+    local vec_selc_stag = {};
```
Comment (raphasampaio) [inline extensions/expression.lua]: keep the variables names expanded, example: vec_selc_stag -> selected_stages

File: extensions/expression.lua [hunk 6]
```diff
@@ -288,3 +288,130 @@ function Expression.add_agents_right(self, ...)
...
+    for stg = inital_stage,last_stage do
+        for _,stg_v in ipairs(vec_stages) do
+            if( stg == stg_v ) then
```
Comment (raphasampaio) [inline extensions/expression.lua]: don't need these parenthesis

File: extensions/expression.lua [hunk 7]
```diff
@@ -288,3 +288,130 @@ function Expression.add_agents_right(self, ...)
...
+    if not self:loaded() then
+        warning(tag .. ": null at " .. PSR.source_line(2));
+        return self;
+    end
+
+    if( data_repli:scenarios() ~= 1) then
+        error("Number of data_repli scenario must be 1")
+    end
+
+    N_scn = self:scenarios();
```
Comment (raphasampaio) [inline extensions/expression.lua]: use local and expanded names

**Review by raphasampaio — CHANGES_REQUESTED** [review]

File: extensions/expression.lua (hunk 1 above)
Comment (Iurysab) [reply extensions/expression.lua]: done

File: extensions/expression.lua (hunk 3 above)
Comment (Iurysab) [reply extensions/expression.lua:309]: done

File: extensions/expression.lua (hunk 2 above)
Comment (Iurysab) [reply extensions/expression.lua]: done

File: extensions/expression.lua (hunk 3 above)
Comment (Iurysab) [reply extensions/expression.lua]: done

File: extensions/expression.lua (hunk 4 above)
Comment (Iurysab) [reply extensions/expression.lua]: done

**Pull request merged**